torch/ao/quantization/fx/_model_report/model_report_visualizer.py

```diff
@@ -526,22 +545,66 @@ def generate_plot_visualization(self, feature_filter: str, module_fqn_filter: st
         # actually show the plot
         plt.show()
 
-    def generate_histogram_visualization(self, feature_filter: str, module_fqn_filter: str = ""):
+    def generate_histogram_visualization(self, feature_filter: str, module_fqn_filter: str = "", num_bins: int = 10):
         r"""
         Takes in a feature and optional module_filter and plots the histogram of desired data.
 
         Note:
             Only features in the report that have tensor value data can be viewed as a histogram
+            If you want to plot a histogram from all the channel values of a specific feature for
+                a specific model, make sure to specify both the model and the feature properly
+                in the filters and you should be able to see a distribution of the channel data
 
         Args:
             feature_filter (str, optional): Filters the features presented to only those that
                 contain this filter substring
                 Default = "", results in all the features being printed
             module_fqn_filter (str, optional): Only includes modules that contains this string
                 Default = "", results in all the modules in the reports to be visible in the table
+            num_bins (int, optional): The number of bins to create the histogram with
+                Default = 10, the values will be split into 10 equal sized bins
 
         Expected Use:
             >>> # displays the histogram
             >>> model_report_visualizer.generate_histogram_visualization(*filters) # displays the histogram
         """
-        pass
+        # checks if we have matplotlib and let's user know to install it if don't
+        if not got_matplotlib:
+            print("make sure to install matplotlib and try again.")
+            return None
+
+        # get the x and y data and if per channel
+        x_data, y_data, data_per_channel = self._get_plottable_data(feature_filter, module_fqn_filter)
+
+        # for histogram, we just care about plotting the y data
+        # plot based on whether data is per channel or not
+        fig = plt.figure()
```

Comment:
would be nice if we make this a separate function as well

Reply:
i think because some of the labels as well as legend, and titles are pretty different from histogram to line plot, it would be better to leave them separate for now.